tool/tctl/common/lock_command.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/kingpin"
+	"github.com/gravitational/trace"
+	"github.com/pborman/uuid"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service"
+)
+
+// LockCommand implements `tctl lock` group of commands.
+type LockCommand struct {
+	config  *service.Config
+	mainCmd *kingpin.CmdClause
+	spec    types.LockSpecV2
+	expires string
+	ttl     time.Duration
+}
+
+// Initialize allows LockCommand to plug itself into the CLI parser.
+func (c *LockCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	c.config = config
+
+	c.mainCmd = app.Command("lock", "Create a new lock.")
+	c.mainCmd.Flag("user", "Name of a Teleport user to disable.").StringVar(&c.spec.Target.User)
+	c.mainCmd.Flag("role", "Name of a Teleport role to disable.").StringVar(&c.spec.Target.Role)
+	c.mainCmd.Flag("login", "Name of a local UNIX user to disable.").StringVar(&c.spec.Target.Login)
+	c.mainCmd.Flag("node", "UUID of a Teleport node to disable.").StringVar(&c.spec.Target.Node)
+	c.mainCmd.Flag("mfa-device", "UUID of a user MFA device to disable.").StringVar(&c.spec.Target.MFADevice)
+	c.mainCmd.Flag("message", "Message to display to locked-out users.").StringVar(&c.spec.Message)
+	c.mainCmd.Flag("expires", "Time point (RFC3339) when the lock expires.").StringVar(&c.expires)
+	c.mainCmd.Flag("ttl", "Time duration after which the lock expires.").DurationVar(&c.ttl)
+}
+
+// TryRun attempts to run subcommands.
+func (c *LockCommand) TryRun(cmd string, client auth.ClientI) (match bool, err error) {
+	ctx := context.TODO()
```

Comment:
I'd be OK with `context.Background()` here - I'm not seeing another place from where it could come from.

Reply:
Well, ideally it should come from `TryRun`'s caller. Notice how the main `Run` function creates its own `context.Background` before calling `TryRun` -- _that_ context should've been passed along. However, as the `TryRun` interface isn't refactored to that end yet, the `tctl` codebase is interspersed with all the `context.TODO`s. I think it's more appropriate to keep it this way to be reminded of the deficiency of the current situation.

Reply:
I'm of the opinion that we should have a single TODO on `Run`, but not plague everywhere else with with context.TODOs - either the refactor happens and it gets sorted out everywhere, or this is the best we can do and that's it. I do get your point, though, so I'll resolve the thread.